projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js

```diff
@@ -63,20 +73,72 @@ export default function Player( { isSelected, attributes } ) {
 		poster,
 	} );
 
-	const { preview, isRequestingEmbedPreview } = usePreview( videoPressUrl );
-	const html = addTokenIntoIframeSource( preview?.html, token );
+	// Helper to invalidate the preview cache.
+	const invalidateResolution = useDispatch( coreStore ).invalidateResolution;
+	const invalidatePreview = useCallback( () => {
+		invalidateResolution( 'getEmbedPreview', [ videoPressUrl ] );
+	}, [ videoPressUrl, invalidateResolution ] );
+
+	const { preview = {}, isRequestingEmbedPreview } = usePreview( videoPressUrl );
+	const html = addTokenIntoIframeSource( preview.html, token );
+
+	const cancelPreviewCheckInterval = useCallback( () => {
+		clearInterval( previewCheckInterval.current );
+	}, [ previewCheckInterval ] );
+
+	// Fetch the preview until it's ready
+	useEffect( () => {
+		// Wait for the client player to load
+		if ( ! isPlayerLoaded ) {
+			return;
+		}
+		previewCheckInterval.current = setInterval( () => {
+			// if the preview is ready or we reached the max attempts, clear the interval
+			if ( isPreviewReady( preview ) || previewCheckAttempts > VIDEO_PREVIEW_ATTEMPTS_LIMIT ) {
+				return cancelPreviewCheckInterval();
+			}
+
+			// Avoid lapping any prior request
+			if ( isRequestingEmbedPreview ) {
+				return;
+			}
+
+			// Invalidate the cache and wait for the next interval since the preview is not ready yet
+			setPreviewCheckAttempts( previewCheckAttempts + 1 );
+			invalidatePreview();
+		}, 1000 );
```

Comment:
Web uses 2 seconds between the requests which we might want to follow. The `setInterval` will run immediately so we get a request at t=0, t=1, etc. I've noticed it takes about 2 attempts which in this set up it takes about a second to get a complete preview.
The other option could be to switch this to a `setTimeout` loop so we get requests at t=1, t=2, etc.